src/routes/blog/post/building-custom-auth-flows/+page.markdoc

```diff
@@ -0,0 +1,544 @@
+    throw new Error('Authentication failed: ' + error.message)
+  }
+}
+```
+
+This function:
+
+1. Sends the login details to our backend
+2. Gets back a token if the login is successful
+3. Creates an Appwrite session with this token
+4. Shows the logged-in screen with the user's name
+
+Next, we add click handlers to our login and logout buttons.
+These functions run when users click the buttons:
+
+```jsx
+// Handle login button click
+document.getElementById('loginButton').addEventListener('click', async () => {
+  const email = document.getElementById('email').value
+  const password = document.getElementById('password').value
+
+  try {
+    await handleExternalAuth(email, password)
+  } catch (error) {
+    alert(error.message)
+  }
+})
+
+// Handle logout
+document.getElementById('logoutButton').addEventListener('click', async () => {
+  try {
+    await account.deleteSession('current')
+    loginForm.classList.remove('hidden')
+    loggedInView.classList.add('hidden')
+  } catch (error) {
+    alert('Logout failed: ' + error.message)
+  }
+})
+```
+
+These click handlers:
+
+- Get the email and password from the form
+- Try to log the user in or out
+- Show error messages if something goes wrong
+- Switch between the login and logged-in screens
+
+Lastly, we check if the user is already logged in when they load the page:
+
+```jsx
+// Check auth status on load
+async function checkAuth() {
+  try {
+    const session = await account.get()
+    loginForm.classList.add('hidden')
+    loggedInView.classList.remove('hidden')
+    userInfo.textContent = `Logged in as: ${session.name}`
+  } catch {
+    loginForm.classList.remove('hidden')
+    loggedInView.classList.add('hidden')
+  }
+}
+
+checkAuth()
+```
+
+This check is important because it:
+
+- Looks for an existing login session
+- Shows the logged-in screen if a session exists
+- Shows the login form if no session is found
+
+# Running the application
+
+To run the application, you'll need to start both the backend and frontend servers.
+First, start the backend server:
+
+```bash
+npm run server
+```
+
+Then, in a new terminal window, start the frontend development server:
+
+```bash
+npm run dev
+```
+
+Navigate to the URL shown by Vite (typically http://localhost:5173) in your browser.
+You can test the authentication using these credentials:
+
+- Email: demo@example.com
+- Password: demo1234
+
+# How it works
+
+The authentication flow in this application follows a specific sequence:
+
+When a user submits the login form, the frontend sends their credentials to our backend server.
+The server then validates these credentials against its user database.
+Upon successful validation, the backend uses Appwrite's SDK to create a custom token for the user.
+
+This token, along with user information, is returned to the frontend.
+The frontend then uses this token to create an Appwrite session, effectively authenticating the user and allowing them to access protected resources.
```

Comment:
Should we explain this section here or before we begin the building part?

Reply:
@adityaoberai That's a valid point to consider, but I intended this to be more of a recap and a bridge to the next section. The how-it-works flow is already naturally explained throughout the implementation

Reply:
> Should we explain this section here or before we begin the building part?

Also, I didn't want to add too much text before the code implementation

Reply:
Fair enough. Could we reframe it like a summary instead? What has been done so far instead of what it does?

I think that would fit the positioning of the paragraph better.

Reply:
@adityaoberai Sounds good. Will do!

Reply:
@adityaoberai Done